src/legacy/ui/public/vis/vis_types/_vislib_vis_legend.scss

```diff
@@ -94,19 +97,45 @@
 
 .visLegend__valueTitle--full {
   word-break: break-all;
+  overflow: hidden;
+  min-width: $colorPickerWidth;
 }
 
 .visLegend__valueDetails {
-  border-bottom: 1px solid $euiColorLightShade;
-  padding-bottom: $euiSizeXS;
+  background-color: $euiColorEmptyShade;
+
+  .visLib--legend-left &,
+  .visLib--legend-right & {
+    margin-top: $euiSizeXS;
+    border-bottom: 1px solid $euiColorLightShade;
+  }
+
+
+  .visLib--legend-top & {
+    position: absolute;
+    margin-top: $euiSizeXS;
+  }
+
+  .visLib--legend-bottom & {
+    position: absolute;
+    margin-bottom: 2 * $euiSizeXS;
+    bottom: $legendLineHeight;
```

Comment:
This doesn't work if the legend wraps:

<img src="https://d.pr/free/i/lxD61Z+" />

Reply:
To fix this, you just need to position the color picker against the `.visLegend__value` instead of the whole list and remove all those unecessary hidden overflows and min-widths:

<img src="https://d.pr/free/i/fFVxZs+" width="50%" />

Reply:
I added the min-width to support multiple open color pickers beside each other:
<img width="614" alt="screenshot 2019-02-15 at 06 04 37" src="https://user-images.githubusercontent.com/1508364/52835903-c3617200-30e8-11e9-93e2-9968eeebdac1.png">

vs

<img width="359" alt="screenshot 2019-02-15 at 06 05 15" src="https://user-images.githubusercontent.com/1508364/52835904-c3fa0880-30e8-11e9-9d9a-1fbe5dadb3c5.png">

If this isn't a feature we depend on, I think it would make sense to only allow one open color picker at a time

Reply:
Yeah I don't usually find that shifting layouts is a good idea, especially with the bug shown here:

![image](https://user-images.githubusercontent.com/549577/52864850-1e0ac600-3109-11e9-8a98-feb801b2069e.png)


I think it's absolutely fine to only show one color picker at a time, if that's a possible solution. Otherwise, I did add in some z-index properties to handle ensuring that if multiple color pickers are open at once, the one that's being hovered on is top most.